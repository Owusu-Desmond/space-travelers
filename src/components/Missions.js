import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addMissions } from '../redux/missions/missions';
import '../styling/Missions.css';

export default function Missions() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(addMissions());
    }
  }, []);

  return (
    <div className="missions-container">
      <div className="row first-row">
        <div>Mission</div>
        <div>Description</div>
        <div>Status</div>
        <div />
      </div>
      {missions.map((mission, index) => (
        <div className={index % 2 === 0 ? 'row lavender' : 'row'} key={mission.id}>
          <div className="col-1">
            <p>{mission.name}</p>
          </div>
          <div className="col-2">
            <p>{mission.description}</p>
          </div>
          <div className="col-3">
            <p>NOT A MEMBER</p>
          </div>
          <div className="col-4">
            <button type="button">Join Mission</button>
          </div>
        </div>
      ))}
    </div>
  );
}
