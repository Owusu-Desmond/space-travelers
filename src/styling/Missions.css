.missions-container {
  margin: 20px 50px;
}

.row {
  display: grid;
  grid-template-columns: 1fr 5fr 1fr 1fr;
  padding: 5px;
  border: 1px solid gray;
}

.first-row,
.col-1 {
  font-size: 20px;
  font-weight: bold;
}

.col-3,
.col-4 {
  display: flex;
  align-items: center;
  justify-content: center;
}

.lavender {
  background-color: lavender;
}

.join-btn,
.leave-btn {
  padding: 5px;
  background-color: inherit;
  border: 2px solid gray;
  border-radius: 4px;
  font-size: 16px;
}

.leave-btn {
  border-color: red;
  color: red;
}

.active-member,
.nota-member {
  background-color: cornflowerblue;
  padding: 5px;
  border-radius: 4px;
  text-align: center;
  color: white;
}

.nota-member {
  background-color: gray;
}
