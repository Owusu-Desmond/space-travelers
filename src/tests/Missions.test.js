import React from "react";
import {rest} from 'msw'
import {setupServer} from 'msw/node'
import {fireEvent, render, screen} from '@testing-library/react'
import { Provider } from "react-redux";
import Missions from '../components/Missions';
import { combineReducers, configureStore } from "@reduxjs/toolkit";
import missionsReducer, { addMissions } from "../redux/missions/missions";
import axios from "axios";


// const server = setupServer(
//   rest.get('https://api.spacexdata.com/v3/missions', (req, res, ctx) => {
//     return res(ctx.json({
//       mission_id: '99',
//       mission_name: 'mission1',
//       description: 'really cool mission',
//     }))
//   })
// )

const rootReducer = combineReducers({
  missions: missionsReducer
})

const store = configureStore({
  reducer: rootReducer,
  preloadedState: {
    missions: [{
      id: '99',
      name: 'mission1',
      description: 'really cool mission',
    }],
  }
});

// // establish API mocking before all tests
// beforeAll(() => server.listen())
// // reset any request handlers that are declared as a part of our tests
// // (i.e. for testing one-time error scenarios)
// afterEach(() => server.resetHandlers())
// // clean up once the tests are done
// afterAll(() => server.close())

it('renders', () => {
  const missions = render(<Provider store={store}><Missions /></Provider>);

  expect(missions).toMatchSnapshot();
})

it('displays state from the store to the UI', () => {
  (<Provider store={store}><Missions /></Provider>);
  screen.debug()

  expect(screen.findByText('really cool mission').toBeInTheDocument)
})
