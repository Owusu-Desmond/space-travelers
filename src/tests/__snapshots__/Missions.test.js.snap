// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="missions-container"
      >
        <div
          class="row first-row"
        >
          <div>
            Mission
          </div>
          <div>
            Description
          </div>
          <div>
            Status
          </div>
          <div />
        </div>
        <div
          class="row lavender"
        >
          <div
            class="col-1"
          >
            <p>
              mission1
            </p>
          </div>
          <div
            class="col-2"
          >
            <p>
              really cool mission
            </p>
          </div>
          <div
            class="col-3"
          >
            <p
              class="nota-member"
            >
              NOT A MEMBER
            </p>
          </div>
          <div
            class="col-4"
          >
            <button
              class="join-btn"
              type="button"
            >
              Join Mission
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="missions-container"
    >
      <div
        class="row first-row"
      >
        <div>
          Mission
        </div>
        <div>
          Description
        </div>
        <div>
          Status
        </div>
        <div />
      </div>
      <div
        class="row lavender"
      >
        <div
          class="col-1"
        >
          <p>
            mission1
          </p>
        </div>
        <div
          class="col-2"
        >
          <p>
            really cool mission
          </p>
        </div>
        <div
          class="col-3"
        >
          <p
            class="nota-member"
          >
            NOT A MEMBER
          </p>
        </div>
        <div
          class="col-4"
        >
          <button
            class="join-btn"
            type="button"
          >
            Join Mission
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
